python/cudf/cudf/tests/test_binops.py

```diff
@@ -1726,3 +1726,49 @@ def decimal_series(input, dtype):
     got = op(a, b)
     assert expect.dtype == got.dtype
     utils.assert_eq(expect, got)
+
+
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "uint8",
+        "uint16",
+        "uint32",
+        "uint64",
+        "int8",
+        "int16",
+        "int32",
+        "int64",
+        "float32",
+        "float64",
+        "str",
+        "datetime64[ns]",
+        "datetime64[us]",
+        "datetime64[ms]",
+        "datetime64[s]",
+        "timedelta64[ns]",
+        "timedelta64[us]",
+        "timedelta64[ms]",
+        "timedelta64[s]",
+    ],
+)
+@pytest.mark.parametrize("null_scalar", [None, cudf.NA])
```

Comment:
Should we support pd.NA and pd.NaT ?

```
>>> s != pd.NA
0    True
1    True
2    True
dtype: bool
>>> s != pd.NaT
0    True
1    True
2    True
dtype: bool
```

Reply:
I think we can support `NaT` at least. What should the behavior be? Currently on branch-0.19, I get:

```
>>> pd.Series([1,2,3], dtype='datetime64[ns]') > np.datetime64('NaT')
0    False
1    False
2    False
dtype: bool
>>> cudf.Series([1,2,3], dtype='datetime64[ns]') > np.datetime64('NaT')
0    <NA>
1    <NA>
2    <NA>
dtype: bool
```
Should we match pandas behavior here since `NaT` isn't the same as `<NA>`? cc @galipremsagar

Reply:
Yup I think we can match pandas behavior because we already do this for `NA` in other columns:

```python
>>> import cudf
>>> s = cudf.Series([1, 2, 3])
>>> s > cudf.NA
0    False
1    False
2    False
dtype: bool
```

But for `nat`, cudf can just continue to treat it as `NA` in these operations too as specified here: https://docs.rapids.ai/api/cudf/nightly/Working-with-missing-data.html#Datetimes

Reply:
Ok - that makes sense. So we just consistently treat `NaT` as `<NA>` everywhere.  My only suggestion is that we make the comparison also return `<NA>` since all comparisons against `<NA>` will return `<NA>` after https://github.com/rapidsai/cudf/issues/7066 is merged.

Reply:
> My only suggestion is that we make the comparison also return `<NA>` since all comparisons against `<NA>` will return `<NA>` after #7066 is merged.

Make sense +1 from my side.